src/Utils/BlockTemplateUtils.php

```diff
@@ -134,25 +134,59 @@ public static function gutenberg_build_template_result_from_post( $post ) {
 	 */
 	public static function gutenberg_build_template_result_from_file( $template_file, $template_type ) {
 		$template_file = (object) $template_file;
+
+		// If the theme has an archive-products.html template but does not have product taxonomy templates
+		// then we will load in the archive-product.html template from the theme to use for product taxonomies on the frontend.
+		$template_is_from_theme = 'theme' === $template_file->source ? true : false;
+		$theme_name             = wp_get_theme()->get( 'Name' );
+
 		// phpcs:ignore WordPress.WP.AlternativeFunctions.file_get_contents_file_get_contents
 		$template_content         = file_get_contents( $template_file->path );
 		$template                 = new \WP_Block_Template();
-		$template->id             = 'woocommerce//' . $template_file->slug;
-		$template->theme          = 'woocommerce/woocommerce';
+		$template->id             = $template_is_from_theme ? strtolower( $theme_name ) . '//' . $template_file->slug : 'woocommerce//' . $template_file->slug;
+		$template->theme          = $template_is_from_theme ? $theme_name : 'woocommerce/woocommerce';
```

Comment:
I think `theme` should be the theme slug, not the name. Ie, when using Twenty Twenty-Two it makes a Rest API request to an endpoint with a space and capital letters:

![imatge](https://user-images.githubusercontent.com/3616980/146569102-f32d90ab-c844-41fa-aa47-e09c0efbd7fb.png)

Reply:
Good catch Ive used `TextDomain` instead of `Name` instead and this fixes this.